samples/sqlservices/src/controllers/mainController.ts

```diff
@@ -188,6 +188,45 @@ export default class MainController implements vscode.Disposable {
 				selectedRow: 2
 			}).component();
 
+			let declarativeTable = view.modelBuilder.declarativeTable()
+			.withProperties({
+				columns: [{
+						displayName: 'Column 1',
+						valueType: sqlops.DeclarativeDataType.string,
+						width: '20px',
+						isReadOnly: true
+					}, {
+						displayName: 'Column 2',
+						valueType: sqlops.DeclarativeDataType.string,
+						width: '100px',
+						isReadOnly: false
+					}, {
+						displayName: 'Column 3',
+						valueType: sqlops.DeclarativeDataType.boolean,
+						width: '20px',
+						isReadOnly: false
+					}, {
+						displayName: 'Column 4',
+						valueType: sqlops.DeclarativeDataType.category,
```

Comment:
>valueType [](start = 6, length = 9)

Should create a work item to support input validation

Reply:
I will

---
In reply to: [194467324](https://github.com/Microsoft/sqlopsstudio/pull/1593#discussion_r194467324) [](ancestors = 194467324)